client_test/rpc_test.go

```diff
@@ -55,24 +55,20 @@ func TestRpcClient(t *testing.T) {
 	defer msgB.Close()
 	defer msgI.Close()
 
-	rpcServerA := rpc.NewRpcServer(&clientA, chainId, createLogger(logDestination, "alice", "server"))
-	rpcClientA, err := rpc.NewRpcClient(rpcServerA.Url(), alice.Address(), chainId, createLogger(logDestination, "alice", "client"))
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	defer rpcServerA.Close()
```

Comment:
Because of the greater encapsulation In introduce, the test no longer defers this function call. I think it is OK, since the server ought to be garbage collected anyway.